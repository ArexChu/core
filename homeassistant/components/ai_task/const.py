"""Constants for the AI Task integration."""

from __future__ import annotations

from enum import StrEnum
from typing import TYPE_CHECKING

from homeassistant.util.hass_dict import HassKey

if TYPE_CHECKING:
    from homeassistant.helpers.entity_component import EntityComponent

    from .entity import AITaskEntity

DOMAIN = "ai_task"
DATA_COMPONENT: HassKey[EntityComponent[AITaskEntity]] = HassKey(DOMAIN)

DEFAULT_SYSTEM_PROMPT = (
    "You are a Home Assistant expert and help users with their tasks."
)


class GenTextTaskType(StrEnum):
    """Generate text task types.

    A task type describes the intent of the request in order to
    match the right model for balance of cost and quality.
    """

    GENERATE = "generate"
    """Generate content, which may target a higher quality result."""

    SUMMARY = "summary"
    """Summarize existing content, which be able to use a more cost effective model."""
