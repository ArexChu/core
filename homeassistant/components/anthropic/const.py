"""Constants for the Anthropic integration."""

import logging

DOMAIN = "anthropic"
LOGGER = logging.getLogger(__package__)

DEFAULT_CONVERSATION_NAME = "Claude conversation"

CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "claude-3-5-haiku-latest"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 3000
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
CONF_THINKING_BUDGET = "thinking_budget"
RECOMMENDED_THINKING_BUDGET = 0
MIN_THINKING_BUDGET = 1024

NON_THINKING_MODELS = [
    "claude-3-5",  # Both sonnet and haiku
    "claude-3-opus",
    "claude-3-haiku",
]
