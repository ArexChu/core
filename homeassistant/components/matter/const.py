"""Constants for the Matter integration."""

import logging

ADDON_SLUG = "core_matter_server"

CONF_INTEGRATION_CREATED_ADDON = "integration_created_addon"
CONF_USE_ADDON = "use_addon"

DOMAIN = "matter"
LOGGER = logging.getLogger(__package__)

# prefixes to identify device identifier id types
ID_TYPE_DEVICE_ID = "deviceid"
ID_TYPE_SERIAL = "serial"

FEATUREMAP_ATTRIBUTE_ID = 65532

SERVICE_SET_UNOCCUPIED_COOLING_TEMPERATURE = "set_unoccupied_cooling_target_temperature"
SERVICE_SET_UNOCCUPIED_HEATING_TEMPERATURE = "set_unoccupied_heating_target_temperature"

ATTR_OCCUPANCY = "occupancy"
ATTR_UNOCCUPIED_COOLING_TARGET_TEMP = "unoccupied_cooling_target_temp"
ATTR_UNOCCUPIED_HEATING_TARGET_TEMP = "unoccupied_heating_target_temp"
