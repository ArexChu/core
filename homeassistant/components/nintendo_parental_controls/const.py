"""Constants for the Nintendo Switch parental controls integration."""

DOMAIN = "nintendo_parental_controls"
CONF_UPDATE_INTERVAL = "update_interval"
CONF_SESSION_TOKEN = "session_token"
