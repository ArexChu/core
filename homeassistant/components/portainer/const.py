"""Constants for the Portainer integration."""

DOMAIN = "portainer"
DEFAULT_NAME = "Portainer"

ENDPOINT_STATUS_DOWN = 2
