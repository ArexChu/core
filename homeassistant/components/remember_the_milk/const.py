"""Constants for the Remember The Milk integration."""

import logging

DOMAIN = "remember_the_milk"
LOGGER = logging.getLogger(__package__)
