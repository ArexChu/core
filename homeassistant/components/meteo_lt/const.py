"""Constants for the Meteo.lt integration."""

from datetime import timedelta

from homeassistant.const import Platform

DOMAIN = "meteo_lt"
PLATFORMS = [Platform.WEATHER]

MANUFACTURER = "Lithuanian Hydrometeorological Service"
MODEL = "Weather Station"

DEFAULT_UPDATE_INTERVAL = timedelta(minutes=30)

CONF_PLACE_CODE = "place_code"

ATTRIBUTION = "Data provided by Lithuanian Hydrometeorological Service (LHMT)"
