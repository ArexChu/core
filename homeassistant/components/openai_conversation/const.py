"""Constants for the OpenAI Conversation integration."""

from dataclasses import dataclass, field
import logging
from typing import Any

from homeassistant.const import CONF_LLM_HASS_API
from homeassistant.helpers import llm

DOMAIN = "openai_conversation"
LOGGER: logging.Logger = logging.getLogger(__package__)

DEFAULT_CONVERSATION_NAME = "OpenAI Conversation"
DEFAULT_AI_TASK_NAME = "OpenAI AI Task"
DEFAULT_NAME = "OpenAI Conversation"

CONF_CHAT_MODEL = "chat_model"
CONF_CODE_INTERPRETER = "code_interpreter"
CONF_FILENAMES = "filenames"
CONF_MAX_TOKENS = "max_tokens"
CONF_PROMPT = "prompt"
CONF_REASONING_EFFORT = "reasoning_effort"
CONF_RECOMMENDED = "recommended"
CONF_TEMPERATURE = "temperature"
CONF_TOP_P = "top_p"
CONF_WEB_SEARCH = "web_search"
CONF_WEB_SEARCH_USER_LOCATION = "user_location"
CONF_WEB_SEARCH_CONTEXT_SIZE = "search_context_size"
CONF_WEB_SEARCH_CITY = "city"
CONF_WEB_SEARCH_REGION = "region"
CONF_WEB_SEARCH_COUNTRY = "country"
CONF_WEB_SEARCH_TIMEZONE = "timezone"
RECOMMENDED_CODE_INTERPRETER = False
RECOMMENDED_CHAT_MODEL = "gpt-4o-mini"
RECOMMENDED_MAX_TOKENS = 3000
RECOMMENDED_REASONING_EFFORT = "low"
RECOMMENDED_TEMPERATURE = 1.0
RECOMMENDED_TOP_P = 1.0
RECOMMENDED_WEB_SEARCH = False
RECOMMENDED_WEB_SEARCH_CONTEXT_SIZE = "medium"
RECOMMENDED_WEB_SEARCH_USER_LOCATION = False

UNSUPPORTED_MODELS: list[str] = [
    "o1-mini",
    "o1-mini-2024-09-12",
    "o1-preview",
    "o1-preview-2024-09-12",
    "gpt-4o-realtime-preview",
    "gpt-4o-realtime-preview-2024-12-17",
    "gpt-4o-realtime-preview-2024-10-01",
    "gpt-4o-mini-realtime-preview",
    "gpt-4o-mini-realtime-preview-2024-12-17",
]

UNSUPPORTED_WEB_SEARCH_MODELS: list[str] = [
    "gpt-3.5",
    "gpt-4-turbo",
    "gpt-4.1-nano",
    "o1",
    "o3-mini",
]


@dataclass(frozen=True)
class ConversationOptions:
    """Configuration options for conversation."""

    recommended: bool = True
    llm_hass_api: list[str] = field(default_factory=lambda: [llm.LLM_API_ASSIST])
    prompt: str = llm.DEFAULT_INSTRUCTIONS_PROMPT
    chat_model: str = RECOMMENDED_CHAT_MODEL
    max_tokens: int = RECOMMENDED_MAX_TOKENS
    temperature: float = RECOMMENDED_TEMPERATURE
    top_p: float = RECOMMENDED_TOP_P
    reasoning_effort: str = RECOMMENDED_REASONING_EFFORT
    web_search: bool = RECOMMENDED_WEB_SEARCH
    code_interpreter: bool = RECOMMENDED_CODE_INTERPRETER

    def to_dict(self) -> dict[str, Any]:
        """Convert to dictionary format."""
        return {
            CONF_RECOMMENDED: self.recommended,
            CONF_LLM_HASS_API: self.llm_hass_api,
            CONF_PROMPT: self.prompt,
            CONF_CHAT_MODEL: self.chat_model,
            CONF_MAX_TOKENS: self.max_tokens,
            CONF_TEMPERATURE: self.temperature,
            CONF_TOP_P: self.top_p,
            CONF_REASONING_EFFORT: self.reasoning_effort,
            CONF_WEB_SEARCH: self.web_search,
            CONF_CODE_INTERPRETER: self.code_interpreter,
        }


@dataclass(frozen=True)
class AITaskOptions:
    """Configuration options for AI tasks."""

    recommended: bool = True

    def to_dict(self) -> dict[str, Any]:
        """Convert to dictionary format."""
        return {CONF_RECOMMENDED: self.recommended}


# Maintain backward compatibility with existing dictionary format
RECOMMENDED_CONVERSATION_OPTIONS = {
    CONF_RECOMMENDED: True,
    CONF_LLM_HASS_API: [llm.LLM_API_ASSIST],
    CONF_PROMPT: llm.DEFAULT_INSTRUCTIONS_PROMPT,
}
RECOMMENDED_AI_TASK_OPTIONS = {
    CONF_RECOMMENDED: True,
}
