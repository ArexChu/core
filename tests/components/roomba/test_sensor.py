"""Tests for IRobotEntity usage in Roomba sensor platform."""

from unittest.mock import AsyncMock, patch

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from tests.common import MockConfigEntry, snapshot_platform


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_entities(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_roomba: AsyncMock,
    snapshot: SnapshotAssertion,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test roomba entities."""
    with patch("homeassistant.components.roomba.PLATFORMS", [Platform.SENSOR]):
        mock_config_entry.add_to_hass(hass)
        await hass.config_entries.async_setup(mock_config_entry.entry_id)
        await hass.async_block_till_done()

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
