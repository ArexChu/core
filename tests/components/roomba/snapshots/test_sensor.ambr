# serializer version: 1
# name: test_entities[sensor.test_roomba_average_mission_time-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_average_mission_time',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Average mission time',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'average_mission_time',
    'unique_id': 'average_mission_time_blid123',
    'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
  })
# ---
# name: test_entities[sensor.test_roomba_average_mission_time-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Average mission time',
      'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_average_mission_time',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_battery-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_battery',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.BATTERY: 'battery'>,
    'original_icon': None,
    'original_name': 'Battery',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'battery_blid123',
    'unit_of_measurement': '%',
  })
# ---
# name: test_entities[sensor.test_roomba_battery-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'battery',
      'friendly_name': 'Test Roomba Battery',
      'unit_of_measurement': '%',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_battery',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_battery_cycles-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_battery_cycles',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Battery cycles',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'battery_cycles',
    'unique_id': 'battery_cycles_blid123',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[sensor.test_roomba_battery_cycles-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Battery cycles',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_battery_cycles',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_canceled_missions-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_canceled_missions',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Canceled missions',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'canceled_missions',
    'unique_id': 'canceled_missions_blid123',
    'unit_of_measurement': 'Missions',
  })
# ---
# name: test_entities[sensor.test_roomba_canceled_missions-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Canceled missions',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'Missions',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_canceled_missions',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_dock_tank_level-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_dock_tank_level',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Dock tank level',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'dock_tank_level',
    'unique_id': 'dock_tank_level_blid123',
    'unit_of_measurement': '%',
  })
# ---
# name: test_entities[sensor.test_roomba_dock_tank_level-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Dock tank level',
      'unit_of_measurement': '%',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_dock_tank_level',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '99',
  })
# ---
# name: test_entities[sensor.test_roomba_failed_missions-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_failed_missions',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Failed missions',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'failed_missions',
    'unique_id': 'failed_missions_blid123',
    'unit_of_measurement': 'Missions',
  })
# ---
# name: test_entities[sensor.test_roomba_failed_missions-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Failed missions',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'Missions',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_failed_missions',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_last_mission_start_time-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_last_mission_start_time',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TIMESTAMP: 'timestamp'>,
    'original_icon': None,
    'original_name': 'Last mission start time',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'last_mission',
    'unique_id': 'last_mission_blid123',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[sensor.test_roomba_last_mission_start_time-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'timestamp',
      'friendly_name': 'Test Roomba Last mission start time',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_last_mission_start_time',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_scrubs-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_scrubs',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Scrubs',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'scrubs_count',
    'unique_id': 'scrubs_count_blid123',
    'unit_of_measurement': 'Scrubs',
  })
# ---
# name: test_entities[sensor.test_roomba_scrubs-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Scrubs',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'Scrubs',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_scrubs',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_successful_missions-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_successful_missions',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Successful missions',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'successful_missions',
    'unique_id': 'successful_missions_blid123',
    'unit_of_measurement': 'Missions',
  })
# ---
# name: test_entities[sensor.test_roomba_successful_missions-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Successful missions',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'Missions',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_successful_missions',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_tank_level-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_tank_level',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Tank level',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'tank_level',
    'unique_id': 'tank_level_blid123',
    'unit_of_measurement': '%',
  })
# ---
# name: test_entities[sensor.test_roomba_tank_level-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Tank level',
      'unit_of_measurement': '%',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_tank_level',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_total_cleaned_area-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_total_cleaned_area',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 0,
      }),
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Total cleaned area',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'total_cleaned_area',
    'unique_id': 'total_cleaned_area_blid123',
    'unit_of_measurement': <UnitOfArea.SQUARE_METERS: 'm²'>,
  })
# ---
# name: test_entities[sensor.test_roomba_total_cleaned_area-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Total cleaned area',
      'unit_of_measurement': <UnitOfArea.SQUARE_METERS: 'm²'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_total_cleaned_area',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_total_cleaning_time-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_total_cleaning_time',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Total cleaning time',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'total_cleaning_time',
    'unique_id': 'total_cleaning_time_blid123',
    'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
  })
# ---
# name: test_entities[sensor.test_roomba_total_cleaning_time-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Total cleaning time',
      'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_total_cleaning_time',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities[sensor.test_roomba_total_missions-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.test_roomba_total_missions',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Total missions',
    'platform': 'roomba',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'total_missions',
    'unique_id': 'total_missions_blid123',
    'unit_of_measurement': 'Missions',
  })
# ---
# name: test_entities[sensor.test_roomba_total_missions-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Roomba Total missions',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'Missions',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.test_roomba_total_missions',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
